src/main/java/com/Logger/domain/repository/InventoryImpl.java

```diff
@@ -0,0 +1,62 @@
+package com.Logger.domain.repository;
+
+import com.mongodb.BasicDBObject;
+import com.Logger.domain.model.Inventory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.aggregation.Aggregation;
+import org.springframework.data.mongodb.core.aggregation.AggregationResults;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.stereotype.Repository;
+import java.util.List;
+import static org.springframework.data.mongodb.core.aggregation.Aggregation.*;
+
+@Repository
+public class InventoryImpl {
+
+    final MongoTemplate mongoTemplate;
+
+    @Autowired
+    public InventoryImpl(MongoTemplate mongoTemplate) {
+        this.mongoTemplate = mongoTemplate;
+    }
+
+    public List<Inventory> findByHotelIdOrRoomIdAndDateRange(String startDate, String endDate, int hotelId, int roomId) {
+        Aggregation agg;
+        if (roomId != 0) {
```

Comment:
use conditional operator instead of if else
 eg:- Aggregation agg = roomId != 0 ? someAggregation() : someOtherAggregation(); just an example